org/postgresql/core/Oid.java

```diff
@@ -83,7 +83,8 @@
     public static String toString(int oid) {
         try {
             Field[] fields = Oid.class.getFields();
-            for (Field field : fields) {
```

Comment:
@davecramer, there is absolutely no performance penalty when using foreach over arrrays (it is compiled to indexed for loop by javac).
There might be a penalty when using foreach for `ArrayLists`, however that depends (JIT can still avoid instantiation of iterator).
Obviously, there is no "penalty" when using foreach over `HashSet` as you anyway have to use iterator.